src/solver/optimisation/opt_optimisation_lineaire.cpp

```diff
@@ -139,6 +139,34 @@ void runThermalHeuristic(PROBLEME_HEBDO* problemeHebdo)
 }
 } // namespace
 
+void resizeProbleme(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
+                    unsigned nombreDeVariables,
+                    unsigned nombreDeContraintes)
+{
+    ProblemeAResoudre->CoutQuadratique.resize(nombreDeVariables);
+    ProblemeAResoudre->CoutLineaire.resize(nombreDeVariables);
+    ProblemeAResoudre->TypeDeVariable.resize(nombreDeVariables);
+    ProblemeAResoudre->Xmin.resize(nombreDeVariables);
+    ProblemeAResoudre->Xmax.resize(nombreDeVariables);
+    ProblemeAResoudre->X.resize(nombreDeVariables);
+    ProblemeAResoudre->AdresseOuPlacerLaValeurDesVariablesOptimisees.resize(nombreDeVariables);
+    ProblemeAResoudre->AdresseOuPlacerLaValeurDesCoutsReduits.resize(nombreDeVariables);
+    ProblemeAResoudre->PositionDeLaVariable.resize(nombreDeVariables);
+    ProblemeAResoudre->Pi.resize(nombreDeVariables);
```

Comment:
nbTermes

Reply:
Pi is initially sized with nombreDeVariables in OPT_AllocateFromNumberOfVariableConstraints.
Why resize it to nbThermes ?

Reply:
`Pi` and `Colonne` are just temporary variables used to fill the constraint matrix. They should not be used after this step, so there is no need to resize them IMO.